cloudant-client/src/main/java/com/cloudant/client/internal/views/ViewResponseImpl.java

```diff
@@ -80,12 +80,12 @@
             totalRows = rows.size();
         }
 
-        long rowsPerPage = (initialQueryParameters.getRowsPerPage() != null) ?
-                initialQueryParameters.getRowsPerPage().longValue() : totalRows;
+        Long rowsPerPage = (initialQueryParameters.getRowsPerPage() != null) ?
+                initialQueryParameters.getRowsPerPage().longValue() : null;
 
-        //we expect limit = rowsPerPage + 1 results, if we have rowsPerPage or less we are on the
-        // last page
-        hasNext = resultRows > rowsPerPage;
+        // We expect limit = rowsPerPage + 1 results, if we have rowsPerPage or less or rowsPerPage
```

Comment:
Comment is now a bit confusing, how about

"// We expect limit = rowsPerPage + 1 results, if we have rowsPerPage or less, or if rowsPerPage wasn't set then we are on the last page"

Reply:
Fixed in c2da7f8